src/coreComponents/physicsSolvers/SolverBase.hpp

```diff
@@ -654,42 +623,29 @@ class SolverBase : public ExecutableGroup
                                  real64 const weakestTol );
 
-
-  template< typename BASETYPE = constitutive::ConstitutiveBase, typename LOOKUP_TYPE >
-  static BASETYPE const & getConstitutiveModel( dataRepository::Group const & dataGroup, LOOKUP_TYPE const & key );
-
-  template< typename BASETYPE = constitutive::ConstitutiveBase, typename LOOKUP_TYPE >
-  static BASETYPE & getConstitutiveModel( dataRepository::Group & dataGroup, LOOKUP_TYPE const & key );
-
   /**
-   * @brief Partially validates constitutive model names input.
-   * @param[in,out] modelNames reference to input array of model names
-   * @param[in] allowEmpty if @p true, empty array is not considered an error
-   * @return flag indicating whether at least one model has been provided
+   * @brief Get the Constitutive Name object
    *
-   * Checks that number of model names is equal to the number of solver's target regions.
-   * Additionally, currently admits a single-element list, which is interpreted as one model
-   * used for all target regions (the list is resized and populated accordingly).
-   * If @p allowEmpty is true and the input is empty, returns false, which the solver can
-   * interpret as a signal this type of model is disabled for the run (for optional models).
+   * @tparam CONSTITUTIVE_BASE_TYPE the base type of the constitutive model.
+   * @param subregion the element subregion on which the constitutive model is registered
+   * @return the name name of the constitutive model of type @p CONSTITUTIVE_BASE_TYPE registered on the @p subregion.
    */
-  bool checkModelNames( array1d< string > & modelNames,
-                        string const & attribute,
-                        bool const allowEmpty = false ) const;
+  template< typename CONSTITUTIVE_BASE_TYPE >
+  static string getConstitutiveName( ElementSubRegionBase const & subRegion );
 
   /**
-   * @brief Populate array of constitutive model indices from list of model names.
-   * @tparam MODEL_TYPE Base class of constitutive models to check against
-   * @param elemRegionManager reference to element manager
-   * @param modelNames list of model names
-   *
-   * This function is typically called from solver's initializePreSubGroups() method,
-   * after constitutive models have been set up but before they are used.
-   * Looks up each model by name and type in each subregion of target regions.
+   * @brief This function sets constitutive name fields on an
+   *  ElementSubRegionBase, and calls the base function it overrides.
+   * @param subRegion The ElementSubRegionBase that will have constitutive
+   *  names set.
    */
-  template< typename MODEL_TYPE = constitutive::ConstitutiveBase >
-  void validateModelMapping( ElementRegionManager const & elemRegionManager,
-                             arrayView1d< string const > const & modelNames ) const;
+  virtual void setConstitutiveNamesCallSuper( ElementSubRegionBase & subRegion ) const { GEOSX_UNUSED_VAR( subRegion ); }
```

Comment:
I must confess that I do not fully understand what you are trying to achieve here. (I know you talked about it during the dev meeting, but I did not catch it...)
It looks like you are willing to enforce the derived class to call the upper function which does nothing?

Reply:
The SolverBase implementation does nothing (for now) but the physicsSolvers hierarchy is deeper. All, flow solvers, for example, will call `FlowSolverBase` implementaiton of `setConstitutiveNamesCallsuper`.

Reply:
What do you expect to do (in the future) with this architecture then?

Reply:
Every time there is a constitutive model with a base time common to all (or many) solvers can be set in the base class. This is already used by the Flow Solvers in which, for example, the solid model is set in `FlowSolveBase` but the fluid model is specific to the solve type and it's set  in the private function.  
I am not sure whether we will ever have something set in `SolverBase` or not. If that's not the case, eventually, we could just make this a pure virtual.